sdk/spring/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/integration/ReactiveEtagIT.java

```diff
@@ -124,4 +124,27 @@ public void testShouldFailIfEtagDoesNotMatch() {
         StepVerifier.create(deleteMono).verifyError(CosmosAccessException.class);
     }
 
+    @Test
+    public void testBulkShouldFailIfEtagDoesNotMatch() {
+        Flux<CourseWithEtag> insertedCourseWithEtagFlux = reactiveCourseWithEtagRepository.saveAll(Flux.just(createCourseWithEtag()));
+        List<CourseWithEtag> insertedCourseWithEtag = insertedCourseWithEtagFlux.collectList().block();
+        Assert.assertEquals(insertedCourseWithEtag.size(), 1);
+
+        insertedCourseWithEtag.get(0).setName("CHANGED");
+        Flux<CourseWithEtag> updatedCourseWithEtagFlux = reactiveCourseWithEtagRepository.saveAll(insertedCourseWithEtag);
+        List<CourseWithEtag> updatedCourseWithEtag = updatedCourseWithEtagFlux.collectList().block();
+        Assert.assertEquals(updatedCourseWithEtag.size(), 1);
+        updatedCourseWithEtag.get(0).setEtag(insertedCourseWithEtag.get(0).getEtag());
+
+
+        Flux<CourseWithEtag> courseFlux = reactiveCourseWithEtagRepository.saveAll(updatedCourseWithEtag);
+        StepVerifier.create(courseFlux);
+        Assert.assertEquals(courseFlux.collectList().block().size(), 0);
```

Comment:
with `StepVerifier` we don't need to use block(), you can test using the reactive calls directly. This is not a blocker, but something to be improved in a follow up PR.

Reply:
I see you have followed the correct usage of `StepVerifier` in the below `ReactiveRoleRepositoryIT` test class.